File: MainThree.java
public class Time {
    private int hour = 0;
    private int minute = 0;
    
    public Time () {
      hour = 0;
      minute = 0;
    }
  
    public Time (int h, int m) {
      if (h >= 1 && h <= 23) {
        hour = h;
      } else {
        hour = 0;
      }
      if (m >= 1 && m <= 59) {
        minute = m;
      } else {
        minute = 0;
      }
    }
    
    void increment() {
      minute++;
      if (minute >= 60) {
        minute = 0;
        hour++;
        if (hour >= 24) {
          hour = 0;
        }
      }
    }
  
    public String convert() {
      String hr = "";
      String min = "";
      
      if (hour == 0) {
        hr = "12";
      } else if (hour < 12) {
        hr = Integer.toString(hour);
      } else if (hour == 12) {
        hr = "12";
      } else {
        hr = Integer.toString(hour - 12);
      }
      
      if (minute < 10) {
        min = "0" + Integer.toString(minute);
      } else {
        min = Integer.toString(minute);
      }
      
      if (hour < 12) {
        return hr + ":" + min + " AM";
      } else {
        return hr + ":" + min + " PM";
      }
    }
  
  
    public String toString () {
      return String.format("%02d%02d", hour, minute); //makes sure each one is 2 digits
    }
  }

class Main {
    public static void main(String[] args) {
      Time t1 = new Time(00, 00);
      System.out.println(t1.convert());
      
      Time t2 = new Time(14, 56);
      System.out.println(t2.convert());
      
      Time t3 = new Time(00, 12);
      System.out.println(t3.convert());
      
      Time t4 = new Time(05, 17);
      System.out.println(t4.convert());
      
      Time t5 = new Time(12, 15);
      System.out.println(t5.convert());
      
      Time t6 = new Time(00, 15);
      System.out.println(t6.convert());
      
      Time t7 = new Time(23, 59);
      System.out.println(t7.convert());
    }
}

File: MainThree.py
class Time:
    def __init__(self, h=0, m=0):
        if 1 <= h <= 23:
            self.hour = h
        else:
            self.hour = 0
        if 1 <= m <= 59:
            self.minute = m
        else:
            self.minute = 0
    
    def increment(self):
        self.minute += 1
        if self.minute >= 60:
            self.minute = 0
            self.hour += 1
            if self.hour >= 24:
                self.hour = 0
    
    def convert(self):
        hr = ""
        min = ""
        if self.hour == 0:
            hr = "12"
        elif self.hour < 12:
            hr = str(self.hour)
        elif self.hour == 12:
            hr = "12"
        else:
            hr = str(self.hour - 12)
        
        if self.minute < 10:
            min = "0" + str(self.minute)
        else:
            min = str(self.minute)
        
        if self.hour < 12:
            return hr + ":" + min + " AM"
        else:
            return hr + ":" + min + " PM"

    def __str__(self):
        return "%02d%02d" % (self.hour, self.minute)

if __name__ == "__main__":
    t1 = Time(0, 0)
    print(t1.convert())

    t2 = Time(14, 56)
    print(t2.convert())

    t3 = Time(0, 12)
    print(t3.convert())

    t4 = Time(5, 17)
    print(t4.convert())

    t5 = Time(12, 15)
    print(t5.convert())

    t6 = Time(0, 15)
    print(t6.convert())

    t7 = Time(23, 59)
    print(t7.convert())
